site/content/entry/yuzu-progress-report-nov-2021/index.md

```diff
@@ -0,0 +1,330 @@
+multiplier of the native framerate as the upper limit.
+
+So for example, if a game runs at 30FPS, and the user wants to limit it to 120FPS to better match the maximum refresh rate of their display, a 4x multiplier would be 
+needed, and a 2x one would do if the game runs at 60FPS natively.
+
+The problem with this approach is that as shown in the example, different games will run at different framerates natively, so a global value just won’t do.
+Kewlan [added support for framerate caps in the custom configuration](https://github.com/yuzu-emu/yuzu/pull/7404), users can access it by right clicking a game in yuzu’s 
+list and selecting properties. This way, the user can set their prefered cap on a per-game basis.
+
+{{< single-title-imgs
+    "If you buy a new display, remember to manually incerase the refresh rate!"
+    "./cap1.png"
+    "./cap2.png"
+    >}}
+
+If the user wants a value that isn’t a multiple of the native framerate of the game, they can use software like 
+[MSI Afterburner](https://www.guru3d.com/files-details/msi-afterburner-beta-download.html) to dial a custom limit.
+
+There has been some extensive work to improve hotkeys.
+
+[v1993](https://github.com/v1993) got a report that yuzu’s window will flicker if the user holds the Esc key when trying to exit fullscreen. 
+[That won’t happen any more](https://github.com/yuzu-emu/yuzu/pull/7353).
+
+german77 discovered that holding a hotkey would constantly spam it, that made him angry, and [this behaviour is no more](https://github.com/yuzu-emu/yuzu/pull/7355).
+
+[heinarmann](https://github.com/heinermann) [replaced the emulation shortcuts and states](https://github.com/yuzu-emu/yuzu/pull/7419):
+
+`Start` has been removed and `Continue` was moved to `Pause` when applicable.
+Exiting the emulator will now work while in fullscreen.
+`Restart` will no longer cause a crash.
+`Load Amiibo` was improved. We still have issues with several games when using this function, so consider it in a beta state.
+`TAS` [got it’s own submenu](https://github.com/yuzu-emu/yuzu/pull/7406) in `Tools > TAS`.
+
+{{< single-title-imgs
+    " "
+    "./tasbug.png"
+    "./tasfix.png"
+    >}}
+
+## Kernel fixes
+
+Morph fixed [a compilation warning](https://github.com/yuzu-emu/yuzu/pull/7278) in the kernel `SVC::WaitSynchronization`, where the parameter "num_handles" was wrongly 
+declared as unsigned integer when it was actually a signed integer.
+
+heinermann discovered a bug in the kernel that crashed yuzu upon exiting.
+
+This happened because dummy threads in the kernel weren't getting destroyed properly. When you exited yuzu during emulation or after stopping emulation, these threads 
+stayed alive even though the kernel got destroyed, thus causing yuzu to crash.
+
+[By making the kernel own these threads](https://github.com/yuzu-emu/yuzu/pull/7359), we ensure they get destroyed when the kernel gets destroyed. 
+Thus the crash is mitigated.
```

Comment:
```suggestion
heinermann discovered a bug in the kernel that crashed yuzu upon exiting.

This happened because dummy threads in the kernel weren't getting destroyed properly. When yuzu was exited during emulation or after stopping emulation, these threads stayed alive even though the kernel got destroyed, thus causing yuzu to crash.

[By making the kernel own these threads](https://github.com/yuzu-emu/yuzu/pull/7359), we ensure they get destroyed when the kernel gets destroyed. 
mitigating the crash.
```

Reply:
Eh what?